scripts/serverless.js

```diff
@@ -296,29 +296,31 @@ processSpanPromise = (async () => {
           }
 
           let envVarNamesNeededForDotenvResolution;
+          resolverConfiguration = {
```

Comment:
This attempt won't work well, as a configuration constructed at this point, is not complete (it's enriched on the way in multiple places, but it's still skipped due to `variablesMeta.size` being `0`).

I've browsed the logic, and from what I see best approach would be to do the following instead:

At https://github.com/serverless/serverless/blob/main/scripts/serverless.js#L647 add:

```javascript
        if (!resolverConfiguration) {
          // There were no variables in the initial configuration, yet it was extended by
          // the plugins with ones.
          // In this case we need to ensure `resolverConfiguration` which initially was not setup
          resolverConfiguration = {
            serviceDir,
            configuration,
            variablesMeta,
            sources: {
              env: require('../lib/configuration/variables/sources/env'),
              file: require('../lib/configuration/variables/sources/file'),
              opt: require('../lib/configuration/variables/sources/opt'),
              self: require('../lib/configuration/variables/sources/self'),
              strToBool: require('../lib/configuration/variables/sources/str-to-bool'),
              sls: require('../lib/configuration/variables/sources/instance-dependent/get-sls')(),
            },
            options: filterSupportedOptions(options, { commandSchema, providerName }),
            fulfilledSources: new Set(['env', 'file', 'self', 'strToBool']),
            propertyPathsToResolve:
              commands[0] === 'plugin'
                ? new Set(['plugin', 'provider\0name', 'provider\0stage', 'useDotenv'])
                : null,
            variableSourcesInConfig,
          };
        }
```

(and there should be no need for a local `isConfigurationExtended` variable, I believe)

Reply:
That seems to work. Done.